addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -499,8 +499,8 @@ var SnippetEditor = publicWidget.Widget.extend({
                         // Consider layout-only elements (like bg-shapes) as empty
                         return el.matches(this.layoutElementsSelector);
                     }));
-                return isEmpty && !$el.hasClass('oe_structure')
-                    && !$el.parent().hasClass('carousel-item')
+                const notRemovableSelector = ".oe_structure, .carousel-item";
+                return isEmpty && !$el[0].matches(notRemovableSelector)
```

Comment:
@bvr-odoo Even though there is the `oe_unremovable` class on the main carousel, I would not trust it (especially since it is the kind of class I want to remove in the future). I would still explicitly check here to not remove the container element of carousel items if present.